primitives/src/circuit/merkle_tree/rescue_merkle_tree.rs

```diff
@@ -166,7 +184,7 @@ pub struct Rescue3AryNodeVar {
     pub is_right_child: BoolVar,
 }
 
-trait MerkleTreeHelperGadget<F: RescueParameter> {
+trait MerkleTreeHelperGadget<F: RescueParameter>: MerkleTreeGadget<RescueMerkleTree<F>, F> {
```

Comment:
this is subtyping relationship is slightly strange. 

What do you think of `MerkleTreeHelperGadget<F: RescueParameter, D: DigestAlgorithmGadget<F>>`?

Reply:
Upon review against [the other PR](https://github.com/EspressoSystems/jellyfish/pull/172/), I've decided to wait until that non-membership proof PR is merged, as the trait definitions will change there anyway. This PR is smaller and can be adapted more easily.

Reply:
I've added a `blocked` label for clarification. Don't wanna duplicate the work to refactor traits here twice.